service/history/replication/executable_task.go

```diff
@@ -430,19 +434,35 @@ func (e *ExecutableTaskImpl) GetNamespaceInfo(
 	namespaceEntry, err := e.NamespaceCache.GetNamespaceByID(namespace.ID(namespaceID))
 	switch err.(type) {
 	case nil:
+		if e.versionedTransition != nil && e.versionedTransition.NamespaceFailoverVersion > namespaceEntry.FailoverVersion() {
+			if !e.ProcessToolBox.Config.EnableReplicationEagerRefreshNamespace() {
+				return "", false, serviceerror.NewInternal(fmt.Sprintf("cannot process task because namespace failover version is not up to date, task version: %v, namespace version: %v", e.versionedTransition.NamespaceFailoverVersion, namespaceEntry.FailoverVersion()))
```

Comment:
Will internal error cause the replication task to be DLQed? 
It feels like we should return an retryable error here as the error will be gone once namespace replication catches up.

Reply:
Yes, InternalError will be retried.

Reply:
it's weird actually. Internal error is meant for non-retryable errors.

Reply:
I was not aware of this convention. Do you have a suggested error type? Or I can create a new one.